### DISCUSSION TITLE
How to remap up and down?

### DISCUSSION THREAD

**Discussion by akhansari** [nvim-neo-tree/neo-tree.nvim#1356] [Q&A] [answered]
Hi,
I want to use "p" and "i" as up and down.
It works, but I get warnings "Invalid mapping for" and "?" doesn't work anymore.

```lua
require("neo-tree").setup({
  window = {
    mappings = {
      ["p"] = "<up>",
      ["i"] = "<down>",
      ...
    },
  },
})
```

**Comment by pysan3** [comment, marked as answer]
The right hand side should point to a name of one of the functions listed here: https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/lua/neo-tree/sources/common/commands.lua, such as "toggle_node". 

That config option does not handle vim's `:nmap` functionality. 

Instead, I believe that you have that keybind set globally (due to a different keyboard layout?). 

If that's the case, you can disable default keybinds with

`["p"] = "noop"`

to not assign any specific keybind in the neo-tree window. 

Does this achieve what you want?

**Reply by akhansari** [reply]
Yes exactly, as the key binding is already set globally, it resolved my problem.
Thank you for your help.

**Comment by 99Percent** [comment]
This is not working for me.

I am using lazyvim.

I remapped "p" like this:
```
keymap.set("n", "p", "<up>", { noremap = false, silent = true })
keymap.set("n", "j", "p", opts)
```
and it the neo-tree config I have remapped p like this
```
    opts = {
      window = {
        position = "right",
        mappings = {
          ["p"] = "noop",
        },
      },
   }
```
I tried noremap = true and also set "p" to "k" with the same result: type `<p>` does nothing in neo-tree when I want to move cursor up (as it does while editing).
